File: Lovecraft/Api/Model/Story.cs
using Lovecraft.Model;

namespace Lovecraft.Api.Model
{
    public class Story
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? CoverUrl { get; set; }
        public string? Summary { get; set; }
        public int? UserId { get; set; }
        public int? TeamId { get; set; }
        public DateTime DateCreated { get; set; }
        public Status Status { get; set; }
        public int? EventId { get; set; }
        public virtual User? User { get; set; }
        public virtual Team? Team { get; set; }
        public virtual Event? Event { get; set; }
        public virtual List<Page> Pages { get; set; }
        public virtual List<StoryVote> StoryVotes { get; set; }
        public virtual List<StoryStoryTag> StoryStoryTags { get; set; }
    }

    public enum Status
    {
        Pending,
        InValidation,
        Online,
        Private,
        Deleted,
        ModerateByAdmin,
        ModerateAuto
    }
}

File: lovecraft-ts/src/schemas/Story.ts
import { Event, Page, Prisma, Status, StoryHistory, StoryTag, StoryVote, Team } from '@prisma/client'
import { IUserDto } from "../schemas/User"

export interface IStoryDto {
  id: number
  title: string
  coverUrl: string
  summary: string
  status: Status
  pages: Page[]
  userId: number
  user: IUserDto
  eventId: number
  event: Event
  teamId: number
  team: Team
  storyVotes: StoryVote[]
  storyTags: StoryTag[]
  storyHistory: StoryHistory | null
  commentCount: number
  createdAt: Date
  updatedAt: Date
}

export type StoryWithRelations = Prisma.StoryGetPayload<{
  include: {
    user: true
    team: true
    event: true
    pages: true
    storyVotes: true
    storyHistories: true
    storyStoryTags: { include: { storyTag: true } }
    _count: {
      select: {
        storyComments: true
      }
    }
  }
}>